### DISCUSSION TITLE
Support Flashrom S29GL064 (SNES Repro) [RESOLVED]

### DISCUSSION THREAD

**Discussion by StanMOI** [sanni/cartreader#988] [Need help]
Hello,

Do you know if listed flash chip  are supported for SNES Repro flashing?

- S29GL064 

- MX29LV640 

Nothing mentioned on [this page ](https://github.com/sanni/cartreader/wiki/Supported-Systems)

Thanks

**Comment by Chomemel** [comment]
29lv640 is for sure. I flash them all the time. 

29gl064 is on the code, so yes. But I've never tested it.

**Reply by StanMOI** [reply]
I soldered one S29GL064 on my own PCB to test it but get error code:
ID Type 1: 9090
ID Type 2: 9090
UNKOWN FLASHROM

**Reply by Chomemel** [reply]
Did you try flashing it with the flashrom adpater too? 
[https://github.com/sanni/cartreader/wiki/Flash-Adapter](url)

**Reply by StanMOI** [reply]
FLASH.ino
Code has line for S29GL064M but not for S29GL064N or "S" version.

```
// S29GL064M
    else if (readByte_Flash(28) == 0x10) {
      println_Msg(F("S29GL064M detected"));
      flashSize = 8388608;
      sectorSize = 65536;
      bufferSize = 32;
    }
    // Unknown S29GL type
    else {
      println_Msg(F("Unknown S29GL Type"));
      flashSize = 4194304;
      sectorSize = 65536;
      bufferSize = 32;
```

**Comment by StanMOI** [comment]
<img width="790" alt="Capture d’écran 2024-07-17 à 14 42 11" src="https://github.com/user-attachments/assets/bc35a781-17b3-45b7-b832-21aa98c74982">

I got Flashrom **S29GL064N90TFI04**
Not sure if i'am doing right 
```
// S29GL064
    } else if ((flashid == 0x017E) || (flashid == 0x0104)) {
    println_Msg(F("S29GL064 detected"));
    println_Msg(FS(ATTENTION_3_3V));
    flashSize = 8388608;
    flashromType = 2;
    //END S29GL064
```

**Comment by StanMOI** [comment]
After update to FW14
Error code:

**ID Type 1: 1CFF
ID Type 2: 1CFF
UNKOWN FLASHROM**

Still not working after changing code line like that:
   ```
 // S29GL064
    } else if (flashid == 0x017E) {
    println_Msg(F("S29GL064 detected"));
    println_Msg(FS(ATTENTION_3_3V));
    flashSize = 8388608;
    flashromType = 2;
    //END S29GL064
```

**Comment by StanMOI** [comment]
I done two PCB, LoROM and HiROM PCB with S29GL064 flashrom.
Here is results of testing with original code.
```
// S29GL064M
    else if (readByte_Flash(28) == 0x10) {
      println_Msg(F("S29GL064M detected"));
      flashSize = 8388608;
      sectorSize = 65536;
      bufferSize = 32;
    }
    // Unknown S29GL type
    else {
      println_Msg(F("Unknown S29GL Type"));
      flashSize = 4194304;
      sectorSize = 65536;
      bufferSize = 32;

```

It's look's that is works for LoROM but freeze with HiROM.
Most be some similar problem which was corrected in version [V13.5](https://github.com/sanni/cartreader/releases/tag/V13.5)
"_SNES: Fix flashing hirom repros with SRAM_"

**LoROM = S29GL064** 
SRAM present on board = 256kbit
Recognize flash = OK
Erase = OK
Write = OK
![IMG_7029](https://github.com/user-attachments/assets/8ad6ae80-1b79-4cd2-86d3-c72c0ef64bbd)

![IMG_7031](https://github.com/user-attachments/assets/6a2cf5ee-6f82-4334-9020-89b87c9ccc60)

![IMG_7026](https://github.com/user-attachments/assets/8a95a2b8-29eb-4b2d-bcab-b3f20c92bffc)

-----------------------
**HiROM = S29GL064** 
SRAM present on board = 256kbit
Recognize flash = Unknown S29GL
Erase = KO
Write = KO
![IMG_7028](https://github.com/user-attachments/assets/65be83f7-a6d4-4da7-aa1f-e6d59e0b9610)

![IMG_7032](https://github.com/user-attachments/assets/721c686e-9f26-43d8-a2be-e975e85e57cb)


MX29LV640 = It's work's I added this flashrom to list of supported flash for SNES Repro flashing into:
[Chinese or custom-made SNES repros](https://github.com/sanni/cartreader/wiki/Supported-Systems)

**Comment by StanMOI** [comment]
I found !
My fault...
On HiROM board I forgot to add "voltage regulator".
With regulator from 5V to 3V, all works fine !
I added Flash S29GL064 to list [Chinese or custom-made SNES repros](https://github.com/sanni/cartreader/wiki/Supported-Systems)

![IMG_7035](https://github.com/user-attachments/assets/9e28c1f8-335e-4301-b888-1c4daad5a843)

![IMG_7033 2](https://github.com/user-attachments/assets/c8936bd7-676b-4761-8d26-e127129df191)

![IMG_7034](https://github.com/user-attachments/assets/4ef28e0f-be70-4de6-9b7d-4dcc5eda9d79)

Thank's anyway to everyone.

**Comment by sanni** [comment]
Just out of curiosity, what happens if you select the "CFI HiROM" option? Does CFI work too?

**Reply by StanMOI** [reply]
CFI writing option, HiROM, all works fine !

![IMG_7036](https://github.com/user-attachments/assets/ae6e4956-26c7-4f9c-ad70-1e32367e4cd0)

**Comment by StanMOI** [comment]
CFI writing option, LoROM = all works fine TOO !
